packages/forklift-console-plugin/src/modules/Providers/views/details/components/DetailsSection/utils/getVSphereProviderWebUILink.ts

```diff
@@ -14,13 +14,11 @@ export const getVSphereProviderWebUILink = (provider: V1beta1Provider): string =
   }
 
   // Calculate link using API URL
-  const url = provider?.spec?.url || '';
-  const suffix = '/sdk';
   const newSuffix = '/ui';
+  const uiLinkRegexp = new RegExp('/sdk(/)*$', 'g');
```

Comment:
a. no need to regexp, we are looking for suffix
b. the regex looks wrong, what am i missing here, what does `(/)*` mean ?

Reply:
This means that only for URLs that are ended with one of the following paths:  `/sdk, /sdk/, sdk///...`,  the web ui will be auto generated by replacing the` /sdk....` with `/ui`
The regexp is convenient for supporting the pattern index and length for the replacement.

Reply:
see comment https://github.com/kubev2v/forklift-console-plugin/pull/1157#discussion_r1606092730

Reply:
actually in this case we only support `/sdk` suffix without trailing `/`

Reply:
OK, so we should support both ``/sdk`` and ``/sdk/`` at least: https://github.com/kubev2v/forklift-console-plugin/blob/f37b33d5746ab07caa9c21723f93b162decb19cb/packages/forklift-console-plugin/src/modules/Providers/utils/validators/provider/vsphere/validateVCenterURL.ts#L34

I'll switch to `"/sdk(/)?$`' instead of` "'/sdk(/)*$'"`

Reply:
i have a concern about using regex here to identify a suffix, because we only want to support `/sdk` and the regex is actually no needed here, using regex makes the code complex and harder to follow.
for example supporting multiple / looks like an error, but because it's a regexp it's hard to immediately see what the error is, e.g. support unsupported suffixes

Reply:
my concern is the complexity of this new code, using regex and complex logic, when we can do what we need using `if...else` , it you are concerned about supporting `/sdk/`we can solve it in ways that keep the code simple, for example removing trailing /`` before testing for `/sdk` or testing for both `/sdk` and `/sdk/` the current suggested code is paying too high price in complexity (readability and complexity) to achieve to small pay ( supporting `/sdk/`)